src/user/user.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import AuthDto from 'src/auth/dto/signupAuth.dto'
+import PrismaService from 'src/prisma/prisma.service'
+import * as argon from 'argon2'
+import { JwtService } from '@nestjs/jwt'
+import { ConfigService } from '@nestjs/config'
+
+@Injectable()
+export default class UserService {
+  constructor(
+    private prisma: PrismaService,
+    private jwt: JwtService,
+    private config: ConfigService,
+  ) {}
+
+  async signup(dto: AuthDto) {
+    const hash = await argon.hash(dto.password)
+
+    const user = await this.prisma.user.create({
+      data: {
+        email: dto.email,
+        password: hash,
+        username: dto.username,
+        fullname: dto.fullname,
+      },
+    })
+
+    // return token
+    return this.signToken(user.id, user.email)
+  }
```

Comment:
Refactor this method for signup to be under the Auth Service but user creation be done from user service. Use dependency injection to access user service from auth service. In this case, we wouldn't need to replicate the sign token functionality and handle the user management and authentication functionalities in a separate manner.